common/lib/util_spi.c

```diff
@@ -367,6 +367,27 @@ uint8_t fw_update(uint32_t offset, uint16_t msg_len, uint8_t *msg_buf, bool sect
 	return FWUPDATE_SUCCESS;
 }
 
+int read_fw_image(uint32_t offset, uint8_t msg_len, uint8_t *msg_buf, uint8_t flash_position)
```

Comment:
Need to check if msg_buf is NULL

Reply:
OK, will check if this parameter is NULL.